src/renderer/atlas/AtlasEngine.r.cpp

```diff
@@ -627,34 +623,27 @@ AtlasEngine::CachedGlyphLayout AtlasEngine::_getCachedGlyphLayout(const wchar_t*
         //   --> offsetY = 0
         //   --> scale   = layoutBox.y / (layoutBox.y + left + right)
         //               = 0.69f
-        offset.x = clampedOverhang.left - clampedOverhang.right;
-        offset.y = clampedOverhang.top - clampedOverhang.bottom;
+        offset.x = std::max(0.0f, overhang.left) - std::max(0.0f, overhang.right);
+        scaleCenter.x = offset.x;
+        scaleCenter.y = _r.fontMetrics.baselineInDIP;
 
-        if ((actualSize.x - layoutBox.x) > _r.dipPerPixel)
+        if ((actualSizeX - layoutBox.x) > _r.dipPerPixel)
         {
             scalingRequired = true;
             offset.x = (overhang.left - overhang.right) * 0.5f;
-            scale.x = layoutBox.x / actualSize.x;
+            scale.x = layoutBox.x / actualSizeX;
             scale.y = scale.x;
+            scaleCenter.x = layoutBox.x * 0.5f;
         }
-        if ((actualSize.y - layoutBox.y) > _r.dipPerPixel)
+        if (overhang.top > _r.dipPerPixel || overhang.bottom > _r.dipPerPixel)

```

Comment:
this is after the overhangs have been sign-adjusted, right?

Reply:
You mean clamped? In that case: No, these are the raw overhangs and a positive value indicates that the glyph clips outside of the layout box we've given it.